src/compiler/scala/tools/nsc/Reporting.scala

```diff
@@ -142,3 +270,334 @@ trait Reporting extends internal.Reporting { self: ast.Positions with Compilatio
+    object OtherMigration extends Other; add(OtherMigration)
+    object OtherMatchAnalysis extends WarningCategory; add(OtherMatchAnalysis)
+    object OtherDebug extends WarningCategory; add(OtherDebug)
+
+    sealed trait WFlag extends WarningCategory { override def summaryCategory: WarningCategory = WFlag }
+    object WFlag extends WFlag { override def includes(o: WarningCategory): Boolean = o.isInstanceOf[WFlag] }; add(WFlag)
+    object WFlagDeadCode extends WFlag; add(WFlagDeadCode)
+    object WFlagExtraImplicit extends WFlag; add(WFlagExtraImplicit)
+    object WFlagNumericWiden extends WFlag; add(WFlagNumericWiden)
+    object WFlagOctalLiteral extends WFlag; add(WFlagOctalLiteral)
+    object WFlagValueDiscard extends WFlag; add(WFlagValueDiscard)
+
+    sealed trait Unused extends WarningCategory { override def summaryCategory: WarningCategory = Unused }
+    object Unused extends Unused { override def includes(o: WarningCategory): Boolean = o.isInstanceOf[Unused] }; add(Unused)
+    object UnusedImports extends Unused; add(UnusedImports)
+    object UnusedPatVars extends Unused; add(UnusedPatVars)
+    object UnusedPrivates extends Unused; add(UnusedPrivates)
+    object UnusedLocals extends Unused; add(UnusedLocals)
+    object UnusedParams extends Unused; add(UnusedParams)
+    object UnusedSilent extends Unused; add(UnusedSilent)
+
+    sealed trait Lint extends WarningCategory { override def summaryCategory: WarningCategory = Lint }
+    object Lint extends Lint { override def includes(o: WarningCategory): Boolean = o.isInstanceOf[Lint] }; add(Lint)
+    object LintAdaptedArgs extends Lint; add(LintAdaptedArgs)
+    object LintNullaryUnit extends Lint; add(LintNullaryUnit)
+    object LintInaccessible extends Lint; add(LintInaccessible)
+    object LintNullaryOverride extends Lint; add(LintNullaryOverride)
+    object LintInferAny extends Lint; add(LintInferAny)
+    object LintMissingInterpolator extends Lint; add(LintMissingInterpolator)
+    object LintDocDetached extends Lint; add(LintDocDetached)
+    object LintPrivateShadow extends Lint; add(LintPrivateShadow)
+    object LintTypeParameterShadow extends Lint; add(LintTypeParameterShadow)
+    object LintPolyImplicitOverload extends Lint; add(LintPolyImplicitOverload)
+    object LintOptionImplicit extends Lint; add(LintOptionImplicit)
+    object LintDelayedinitSelect extends Lint; add(LintDelayedinitSelect)
+    object LintPackageObjectClasses extends Lint; add(LintPackageObjectClasses)
+    object LintStarsAlign extends Lint; add(LintStarsAlign)
+    object LintConstant extends Lint; add(LintConstant)
+    object LintNonlocalReturn extends Lint; add(LintNonlocalReturn)
+    object LintImplicitNotFound extends Lint; add(LintImplicitNotFound)
+    object LintSerial extends Lint; add(LintSerial)
+    object LintEtaZero extends Lint; add(LintEtaZero)
+    object LintEtaSam extends Lint; add(LintEtaSam)
+    object LintDeprecation extends Lint; add(LintDeprecation)
```

Comment:
Is there an easy mechanism to reduce redundancy? Easy in the sense of impossible is doable and difficult is easy.

Reply:
This is the best I could come up with.. We cannot put `add(this)` it in the constructor of `WarningCategory`, as the objects here are evaluated lazily.